tasks/main.yml

```diff
@@ -54,3 +54,18 @@
   when: not nginx_allow_shadow
   changed_when: nginx_allow_shadow_remove.rc != 6
   failed_when: nginx_allow_shadow_remove.rc != 6 and nginx_allow_shadow_remove.rc != 0
+
+- name: Crate script directory
+  file:
+    path: /etc/nginx/scripts.d
+    state: directory
+
+- name: Copy scripts
+  copy:
+    src: "{{ item }}"
+    dest: /etc/nginx/scripts.d/{{ item }}
+    mode: 0755
+    owner: root
+    group: root
+  with_items:
+    - ip.sh
```

Comment:
Sicher dass das ein guter Name ist? Ist schon etwas allgemein. Außerdem, wäre es nicht sinnvoller, das mit iptables zu machen statt nginx? (ich weiß nicht, vielleicht ists mit nginx auch sinnvoller…)

Reply:
Der Name ist definitiv nicht "optimal". Kann mir gernen einen sinnvollere ausdenken.

An sich würde ich iptables auch bevorzugen, aber das ist bei Verwendung eines Reverse Proxys keine Option. Alle eingehenden Pakete haben die IP vom Reverse Proxy. Die ursprüngliche Host IP wird durch das x-forwarded-for Feld im HTTP Header gekennzeichnet. Da aber HTTP Layer 5 ist, kann man bei iptables da keine Regel für definieren. In dem Fall muss man dann die IP Adressen über nginx blockieren.

Reply:
können wir nicht fail2ban auf dem reverseproxy ausrollen mit iptables und den logs von z.b cms01?

Reply:
Das könnte man vielleicht schon tun, nur wird das halt gleich viel komplexer als fail2ban lokal auf verschiedenen Hosts zu verwenden weil die dann miteinander kommunizieren müssen. Ich weiß erlich gesagt auch nicht ob Fail2ban standardmäßig ein Backend unterstützt, dass das Lesen mehrerer Dateien (die auch noch auf anderen Servern liegen) unterstützt. Im Moment ist Fail2ban so konfiguriert, dass das Journald ausliest.

Das Bannen von IP Adressen über Nginx ist auch nur bei HTTP-Hosts erforderlich die hinter dem Reverse Proxy hängen. Bei allen anderen Diensten, z.B. SSH kann man einfach über IPTables bannen.